inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl

```diff
@@ -159,7 +180,17 @@ KERNEL(pooling_gpu)(
                                 arg_max_idx = input_idx_bfyx_no_padding;
                             }
 #endif
-                            result = FUNC_CALL(apply_pooling)(result, TO_ACCUMULATOR_TYPE(input[input_idx]));
+                            const ACCUMULATOR_TYPE casted_input = TO_ACCUMULATOR_TYPE(input[input_idx]);
+                            #ifdef SECOND_OUTPUT_TYPE
+                                if (casted_input > result)
+                                {
+                                    result = casted_input;
+                                    result_idx = input_idx;
+                                }
+                            #else
+                                result = FUNC_CALL(apply_pooling)(result, casted_input);
+                            #endif
+
```

Comment:
MaxPool-8 should be implemented for `CHECK_BOUNDARY == 0` case as well. You can add a test for MaxPool-8 where feature map spatial dimensions are evenly divisible by kernel size to validate this case.